src/providers/ldap/sdap_async_groups.c

```diff
@@ -274,10 +275,21 @@ static int sdap_fill_memberships(struct sdap_options *opts,
         if (ghosts == NULL) {
             hret = HASH_ERROR_KEY_NOT_FOUND;
         } else {
+            if (tmpctx == NULL) {
+                /* allocate pool for trimming dn based on first dn length */
+                /* but at least 256 bytes */
+                tmpctx = talloc_pool(NULL,
+                                     5 * (strlen((char *)values[i].data)) + 256);
```

Comment:
`5*first+256` seems to be quite random, but I understand there is no trivial way to have a better estimation.